doc/tutorial/mutli_arch.hpp

```diff
@@ -0,0 +1,158 @@
+Once you are acquainted with dynamic loading, let's start building our own SIMD dynamic system.
+
+## Writing the kernel
+Let's write a small function that will consume a pointer to a data block containing `float` values
+and the size of the data block to process. Inside this kernel, we'll use some algorithms to
+vectorise this process.
+
+@snippet test/integration/multi-arch/kernel.cpp kernel
+
+The code breaks down this way:
+  + We start by including all the necessary include files for the algorithm, functions and types we
+    will use.
+  + We define the `compute_kernel` function. This function takes a pointer and a size as parameters
+    and is declared `extern "C"` to prevent unwanted C++ mangling of the symbol. This is a classical
+    process for dynamic library. Remark we use the `"compute_kernel"` symbol directly. This is possible
+    because the `compute_kernel` function has been compiled with `extern "C"`, thus preventing
+    C++ mangling to occur.
+  + Inside `compute_kernel`, we display some informations about the SIMD API we will be using and
+    proceed to the computation. As a quick way to handle the data, we pass a `std::span` to our
+    algorithm.
+
+Nothing special is required except for the `extern "C"` attribute. If we want to get around this
+limitation and have arbitrary function taking arbitrary C++ types as parameters, there is different
+strategies discussed in various place from arbitrarily use the mangled name to just export a function
+returning an array or a structure containing all the pre-computed function pointer from the .so.
+
+## Writing the dynamic function hub
+We now need to write an actual `compute` function that our users will call. This is the place where
+most of the OS dependent code will be written. The example we give is made to work on Linux. For a
+more OS independent way to handle dynamic loading, you can give a shot at
+[libltdl](https://www.gnu.org/software/libtool/manual/html_node/Using-libltdl.html) or use your
+preferred OS API.
+
+The code looks like this:
+
+@snippet test/integration/multi-arch/compute.cpp compute
+
+What is going on inside `compute` ?
+  + We first open the correct dynamic library depending on the actual SIMD support detected at
+    runtime via eve::is_supported. Notice the names of the .so files, we will go back to those
+    afterward but understand that they are arbitrary. To do so, we use *Immediately Invoked Lambda
+    Expressions* to perform this initialisation.
+  + Next, we load the `compute_kernel` symbol from the handle we just loaded. As opening and fetching
+    pointer to symbol can be costly, we will make it so those tasks are done once by storing both
+    results in static variables.
+  + Once everything is loaded and setup, we call the function pointer with the appropriate parameters.
+
+Obviously, in realistic settings you would actually care about runtime issues, check that every pointers
```

Comment:
"every pointers are non-null" -> "every pointer is non-null" for subject/verb agreement